File: Languages/PHP/Day025.php
<!-- Languages - PHP - Day 25 -->
<!-- Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables. -->

<!-- Create a function that converts a Decimal Number to Binary without using built-in conversion functions.-->


<?php

function decimalToBinary($number) {
    if ($number == 0) {
        return "0";
    }

    $num = abs($number);
    $binaryParts = [];

    while ($num > 0) {
        array_unshift($binaryParts, $num % 2);
        $num = intdiv($num, 2);
    }

    $binary = implode("", $binaryParts);
    return $number < 0 ? "-" . $binary : $binary;
}

echo "Enter a decimal number: ";
$input = trim(fgets(STDIN));

if (is_numeric($input) && strpos($input, '.') === false) {
    $number = (int)$input;
    $binary = decimalToBinary($number);
    echo "Binary representation: $binary\n";
} else {
    echo "Invalid input. Please enter a valid integer.\n";
}

?>

File: Languages/R/Day025.r
# Languages - R - Day 25
# Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

# Create a function that converts a Decimal Number to Binary without using built-in conversion functions.


decimal_to_binary <- function(number) {
  if (number == 0) {
    return("0")
  }

  num <- abs(number)
  binary <- ""
  
  while (num > 0) {
    binary <- paste0(num %% 2, binary)
    num <- num %/% 2
  }

  if (number < 0) {
    binary <- paste0("-", binary)
  }

  return(binary)
}

cat("Enter a decimal number:\n")
input <- trimws(readLines("stdin", n = 1))

if (grepl("^-?\\d+$", input)) {
  number <- as.integer(input)
  binary <- decimal_to_binary(number)
  cat("Binary representation:", binary, "\n")
} else {
  cat("Invalid input. Please enter a valid integer.\n")
}
